### ISSUE TITLE
Size and direction options ordering swallow direction setting when ordered alphabetically

### ISSUE THREAD

**Issue by yamafaktory** [akinsho/toggleterm.nvim#105]
Hi @akinsho and thanks for this awesome project 👍️!

While playing with the configuration, I noticed a somewhat surprising behavior.

Depending on how you order the configuration options, the `direction` one might be skipped.

# Working configuration

In this case the direction is working as expected:

```lua
use {
  'akinsho/toggleterm.nvim',
  config = function()
    require('toggleterm').setup {
      size = 70,
      direction = 'vertical',
      float_opts = {
        border = 'none'
      },
    }
  end
}
```

# Non-working configuration

In this case the direction fallbacks to `horizontal`:

```lua
use {
  'akinsho/toggleterm.nvim',
  config = function()
    require('toggleterm').setup {
      direction = 'vertical',
      size = 70,
      float_opts = {
        border = 'none'
      },
    }
  end
}
```

TL;DR: `size` must be set before `direction` to get `direction` working.

**Comment by akinsho** [comment]
I've just revisited this for the first time in a while and I don't believe this is the case (at least anymore). I've just tried moving the size key and it doesn't change whether or not the direction key works (on the latest commit of this repo).

If you're still experiencing this issue let me know on what version and also a minimal init.lua that I can use to test with
